src/vehicle.cpp

```diff
@@ -225,6 +225,18 @@ vehicle::~vehicle()
 {
 }
 
+void vehicle::set_hp( vehicle_part &pt, int qty )
+{
+    double k = pt.base.max_damage() / double( pt.info().durability );
+    pt.base.set_damage( pt.base.max_damage() - ( qty * k ) );
+}
+
+bool vehicle::mod_hp( vehicle_part &pt, int qty, damage_type dt )
+{
+    double k = pt.base.max_damage() / double( pt.info().durability );
+    return pt.base.mod_damage( - qty * k, dt );
```

Comment:
`mod_hp` invokes `item::on_damage` whereas `set_hp` doesn't. `mod_hp` also returns if any item should be destroyed.